train.py

```diff
@@ -176,6 +178,8 @@ def train(hyp,  # path/to/hyp.yaml or hyp dictionary
 
         # EMA
         if ema and ckpt.get('ema'):
+            if not isinstance(ckpt['ema'].model[-1].anchor_grid, list): # compatibility for new Detect layer
+                ckpt['ema'].model[-1].anchors *= ckpt['ema'].model[-1].stride.view(-1, 1, 1)
```

Comment:
I'm not sure if this change is needed. Couldn't find a test-case which uses `ema`. As it also loads weights from checkpoint, I added the compatibility code here as well.

Reply:
If models are trained using this PR then the EMA will already include anchors multiplied by stride I think, and this multiplication will cause problems as it will multiply the anchors again by stride.

Reply:
Not needed in new refactoring.